process/factory/shard/preProcessorsContainerFactory.go

```diff
@@ -177,33 +178,50 @@ func (ppcf *preProcessorsContainerFactory) createTxPreProcessor() (process.PrePr
 	return preprocess.NewTransactionPreprocessor(args)
 }
 
+// SmartContractResultPreProcessorCreator defines the interface of a smart contract result pre-processor creator
+type SmartContractResultPreProcessorCreator interface {
```

Comment:
Could be this moved in process/interface.go or process/factory/interface.go or process/factory/shard/interface.go ?

Reply:
moved to process/block/preprocess/interface.go to not have cyclic dependency